pkg/channel/lookup.go

```diff
@@ -82,28 +82,33 @@ func DetermineTemplatesVisibility(kyma *v1beta2.Kyma, templates ModuleTemplatesB
 	}
 }
 
-func CheckForOutdatedTemplates(logger logr.Logger, kyma *v1beta2.Kyma, templates ModuleTemplatesByModuleName) {
+func CheckValidTemplatesUpdate(logger logr.Logger, kyma *v1beta2.Kyma, templates ModuleTemplatesByModuleName) {
 	// in the case that the kyma spec did not change, we only have to verify
 	// that all desired templates are still referenced in the latest spec generation
 	for moduleName, moduleTemplate := range templates {
 		moduleTemplate := moduleTemplate
 		for i := range kyma.Status.Modules {
 			moduleStatus := &kyma.Status.Modules[i]
-			if moduleStatus.FQDN == moduleName && moduleTemplate.ModuleTemplate != nil {
-				CheckForOutdatedTemplate(logger, &moduleTemplate, moduleStatus)
+			if moduleMatch(moduleStatus, moduleName) && moduleTemplate.ModuleTemplate != nil {
+				CheckValidTemplateUpdate(logger, moduleTemplate, moduleStatus)
 			}
 		}
+		templates[moduleName] = moduleTemplate
 	}
 }
 
-// CheckForOutdatedTemplate verifies if the given ModuleTemplate is outdated and sets their Outdated Flag based on
-// provided Modules, provided by the Cluster as a status of the last known module state.
+func moduleMatch(moduleStatus *v1beta2.ModuleStatus, moduleName string) bool {
+	return moduleStatus.FQDN == moduleName || moduleStatus.Name == moduleName
```

Comment:
Not sure if this is correct and we should rather compare FQDN with FQDN and name with name, but rest lgtm.